src/uwtools/config/support.py

```diff
@@ -8,6 +9,16 @@
 INCLUDE_TAG = "!INCLUDE"
 
 
+def config_sections(config: configparser.ConfigParser):
+    """
+    Access the _sections method of a config object.
```

Comment:
`:param` entry for `config`, `:return` entry, and `->` return type on function signature, please.

Reply:
I was pondering this one, the _section method is kind of weird in terms of type

Reply:
Hrm, its a 'dictionary of key types', which is giving the error `expected "SupportsKeysAndGetItem[object, str]"` when I try to type it, which isn't terribly helpful. [This](https://stackoverflow.com/questions/68907243/a-dictionary-with-a-given-key-type-cannot-be-assigned-as-a-dictionary-of-objects) StackOverflow entry suggests it may be a covariance issue?

Reply:
Maybe just ignore this for now, then. I may do a small PR to replace this with something that doesn't try to use private methods to get what we want.

Reply:
I had an alternate version for a minute that went straight to pulling the `dict` using the open `sections()` method but it was causing more issues and went back to this one. I'll see if I can come up with a solution too

Reply:
Since INI is a fixed number of levels, we don't even need recursion to produce a real `dict` from a `ConfigParser` object:
```
>>> from configparser import ConfigParser
>>> c = ConfigParser()
>>> c.read_string("[fruit]\na=apple\nb=banana\n")
>>> {s: {k: v for k, v in c[s].items()} for s in c.sections()}
{'fruit': {'a': 'apple', 'b': 'banana'}}
```
Using that should negate the need for those `pylint` and `type` suppressors, should make the type annotation trivial (`-> Dict[str, Dict[str, str]]` I think), and should simplify things on the calling side, too: Create the `ConfigParser`, read into it, then convert it to a `dict`.

Reply:
As I commented elsewhere, this entire function could easily disappear by simplifying the call sites and inlining simple `ConfigParser` -> `dict` expressions there.